internal/user/repository/eventsourcing/eventstore.go

```diff
@@ -0,0 +1,760 @@
+	}
+	existing, err := es.UserByID(ctx, userID)
+	if err != nil {
+		return err
+	}
+	if existing.EmailCode == nil {
+		return caos_errs.ThrowNotFound(nil, "EVENT-lso9w", "code not found")
+	}
+	if err := crypto.VerifyCode(existing.EmailCode.CreationDate, existing.EmailCode.Expiry, existing.EmailCode.Code, verificationCode, es.EmailVerificationCode); err != nil {
+		return err
+	}
+
+	repoExisting := model.UserFromModel(existing)
+	updateAggregate := EmailVerifiedAggregate(es.AggregateCreator(), repoExisting)
+	err = es_sdk.Push(ctx, es.PushAggregates, repoExisting.AppendEvents, updateAggregate)
+	if err != nil {
+		return err
+	}
+
+	es.userCache.cacheUser(repoExisting)
+	return nil
+}
+
+func (es *UserEventstore) CreateEmailVerificationCode(ctx context.Context, userID string) error {
+	if userID == "" {
+		return caos_errs.ThrowPreconditionFailed(nil, "EVENT-lco09", "userID missing")
+	}
+	existing, err := es.UserByID(ctx, userID)
+	if err != nil {
+		return err
+	}
+	repoExisting := model.UserFromModel(existing)
+	if repoExisting.Email == nil {
+		return caos_errs.ThrowPreconditionFailed(nil, "EVENT-pdo9s", "no email existing")
+	}
+	if repoExisting.IsEmailVerified {
+		return caos_errs.ThrowPreconditionFailed(nil, "EVENT-pdo9s", "email already verified")
+	}
+
+	emailCode := new(model.EmailCode)
+	emailCodeCrypto, _, err := crypto.NewCode(es.EmailVerificationCode)
+	if err != nil {
+		return err
+	}
+	emailCode.Code = emailCodeCrypto
+	emailCode.Expiry = es.EmailVerificationCode.Expiry()
+
+	updateAggregate := EmailVerificationCodeAggregate(es.AggregateCreator(), repoExisting, emailCode)
+	err = es_sdk.Push(ctx, es.PushAggregates, repoExisting.AppendEvents, updateAggregate)
+	if err != nil {
+		return err
+	}
+
+	es.userCache.cacheUser(repoExisting)
+	return nil
+}
+
+func (es *UserEventstore) PhoneByID(ctx context.Context, userID string) (*usr_model.Phone, error) {
+	if userID == "" {
+		return nil, caos_errs.ThrowPreconditionFailed(nil, "EVENT-do9se", "userID missing")
+	}
+	user, err := es.UserByID(ctx, userID)
+	if err != nil {
+		return nil, err
+	}
+
+	if user.Phone != nil {
+		return user.Phone, nil
+	}
+	return nil, caos_errs.ThrowNotFound(nil, "EVENT-pos9e", "phone not found")
+}
+
+func (es *UserEventstore) ChangePhone(ctx context.Context, phone *usr_model.Phone) (*usr_model.Phone, error) {
+	if !phone.IsValid() {
+		return nil, caos_errs.ThrowPreconditionFailed(nil, "EVENT-do9s4", "phone is invalid")
+	}
+	existing, err := es.UserByID(ctx, phone.AggregateID)
+	if err != nil {
+		return nil, err
+	}
+	repoExisting := model.UserFromModel(existing)
+	repoNew := model.PhoneFromModel(phone)
+
+	var phoneCode *model.PhoneCode
+	if !phone.IsPhoneVerified {
+		phoneCode = new(model.PhoneCode)
+		phoneCodeCrypto, _, err := crypto.NewCode(es.PhoneVerificationCode)
+		if err != nil {
+			return nil, err
+		}
+		phoneCode.Code = phoneCodeCrypto
+		phoneCode.Expiry = es.PhoneVerificationCode.Expiry()
+	}
```

Comment:
wouldn't it make sense to create a separate method for this?

Reply:
i made separat code for code gen